### DISCUSSION TITLE
Ubuntu 24.04 apt upgrade trying to install grub - should I be worried?

### DISCUSSION THREAD

**Discussion by almightiest** [Sithuk/ubuntu-server-zfsbootmenu#51] [General]
I'd really like to avoid borking my system.  Should I be worried?  I couldn't find a way around not installing grub here, but maybe I could do some kind of package blacklisting that's not included in the install script here?

```
$ sudo apt upgrade -V
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
The following NEW packages will be installed:
   grub-common (2.12-1ubuntu7)
   grub-gfxpayload-lists (0.7build2)
   grub-pc (2.12-1ubuntu7)
   grub-pc-bin (2.12-1ubuntu7)
   grub2-common (2.12-1ubuntu7)
   linux-headers-6.8.0-36 (6.8.0-36.36)
   linux-headers-6.8.0-36-generic (6.8.0-36.36)
   linux-image-6.8.0-36-generic (6.8.0-36.36)
   linux-modules-6.8.0-36-generic (6.8.0-36.36)
   linux-modules-extra-6.8.0-36-generic (6.8.0-36.36)
   os-prober (1.81ubuntu4)
The following packages will be upgraded:
   linux-headers-generic (6.8.0-35.35 => 6.8.0-36.36)
   linux-image-generic (6.8.0-35.35 => 6.8.0-36.36)
2 upgraded, 11 newly installed, 0 to remove and 0 not upgraded.
Need to get 189 MB of archives.
After this operation, 298 MB of additional disk space will be used.
Do you want to continue? [Y/n] n
Abort.
```

**Comment by almightiest** [comment]
This [GitHub gist](https://gist.github.com/firepol/adca8ecc331d4c5ff1bd220ff35e5256) shows how to disable grub packages from apt.  Should this be added into the script?

I created `/etc/apt/preferences.d/grub-prevent-install` and added the gist in there, ran `apt update && apt upgrade` and grub was no longer trying to install.

**Comment by vitali-fridman** [comment]
I've done several upgrades from 22.04 and 23.10 to 24.04. I have seen these traces about grub during the upgrade. They are completely harmless, you will still end up with ZBM as before the upgrade.  One think to note is that your pyznap installation might get out of order due to incompatibilities between upgraded python and modules installed for pyznap. There is another thread were I shared my experience in how I dealt with it.
